### ISSUE TITLE
Object detection training is not starting as given for coral edge tpu

### ISSUE THREAD

**Issue by avinash00721** [tensorflow/models#7024]
Please go to Stack Overflow for help and support:

http://stackoverflow.com/questions/tagged/tensorflow

Also, please understand that many of the models included in this repository are experimental and research-style code. If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **What is the top-level directory of the model you are using**:
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **TensorFlow installed from (source or binary)**:
- **TensorFlow version (use command below)**:
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

`python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by avinash00721** [comment]
sdataset.txt
+ grep american_bulldog /tensorflow/models/research/learn_pet/pet/annotations/test_petsdataset.txt
+ echo 'PREPARING label map...'
PREPARING label map...
+ cd /tensorflow/models/research
+ cp object_detection/data/pet_label_map.pbtxt /tensorflow/models/research/learn_pet/pet
+ echo 'CONVERTING dataset to TF Record...'
CONVERTING dataset to TF Record...
+ python object_detection/dataset_tools/create_pet_tf_record.py --label_map_path=/tensorflow/models/research/learn_pet/pet/pet_label_map.pbtxt --data_dir=/tensorflow/models/research/learn_pet/pet --output_dir=/tensorflow/models/research/learn_pet/pet
/tensorflow/models/research/object_detection/utils/dataset_util.py:75: FutureWarning: The behavior of this method will change in future versions. Use specific 'len(elem)' or 'elem is not None' test instead.
  if not xml:
WARNING:root:Could not find /tensorflow/models/research/learn_pet/pet/annotations/xmls/Abyssinian_104.xml, ignoring example.
root@a49b87cf97db:/tensorflow/models/research# NUM_TRAINING_STEPS=500 && \
> NUM_EVAL_STEPS=100
root@a49b87cf97db:/tensorflow/models/research# clear

root@a49b87cf97db:/tensorflow/models/research# ./retrain_detection_model.sh \
> --num_training_steps ${NUM_TRAINING_STEPS} \
> --num_eval_steps ${NUM_EVAL_STEPS}
+ num_training_steps=500
+ [[ 4 -gt 0 ]]
+ case "$1" in
+ num_training_steps=500
+ shift 2
+ [[ 2 -gt 0 ]]
+ case "$1" in
+ num_eval_steps=100
+ shift 2
+ [[ 0 -gt 0 ]]
+ source /tensorflow/models/research/constants.sh
++ declare -A ckpt_link_map
++ declare -A ckpt_name_map
++ declare -A config_filename_map
++ ckpt_link_map["mobilenet_v1_ssd"]=http://download.tensorflow.org/models/object_detection/ssd_mobilenet_v1_quantized_300x300_coco14_sync_2018_07_18.tar.gz
++ ckpt_link_map["mobilenet_v2_ssd"]=http://download.tensorflow.org/models/object_detection/ssd_mobilenet_v2_quantized_300x300_coco_2019_01_03.tar.gz
++ ckpt_name_map["mobilenet_v1_ssd"]=ssd_mobilenet_v1_quantized_300x300_coco14_sync_2018_07_18
++ ckpt_name_map["mobilenet_v2_ssd"]=ssd_mobilenet_v2_quantized_300x300_coco_2019_01_03
++ config_filename_map["mobilenet_v1_ssd-true"]=pipeline_mobilenet_v1_ssd_retrain_whole_model.config
++ config_filename_map["mobilenet_v1_ssd-false"]=pipeline_mobilenet_v1_ssd_retrain_last_few_layers.config
++ config_filename_map["mobilenet_v2_ssd-true"]=pipeline_mobilenet_v2_ssd_retrain_whole_model.config
++ config_filename_map["mobilenet_v2_ssd-false"]=pipeline_mobilenet_v2_ssd_retrain_last_few_layers.config
++ INPUT_TENSORS=normalized_input_image_tensor
++ OUTPUT_TENSORS=TFLite_Detection_PostProcess,TFLite_Detection_PostProcess:1,TFLite_Detection_PostProcess:2,TFLite_Detection_PostProcess:3
++ OBJ_DET_DIR=/tensorflow/models/research
++ LEARN_DIR=/tensorflow/models/research/learn_pet
++ DATASET_DIR=/tensorflow/models/research/learn_pet/pet
++ CKPT_DIR=/tensorflow/models/research/learn_pet/ckpt
++ TRAIN_DIR=/tensorflow/models/research/learn_pet/train
++ OUTPUT_DIR=/tensorflow/models/research/learn_pet/models
+ mkdir /tensorflow/models/research/learn_pet/train
+ python object_detection/model_main.py --pipeline_config_path=/tensorflow/models/research/learn_pet/ckpt/pipeline.config --model_dir=/tensorflow/models/research/learn_pet/train --num_train_steps=500 --num_eval_steps=100
/tensorflow/models/research/object_detection/utils/visualization_utils.py:25: UserWarning: 
This call to matplotlib.use() has no effect because the backend has already
been chosen; matplotlib.use() must be called *before* pylab, matplotlib.pyplot,
or matplotlib.backends is imported for the first time.

The backend was *originally* set to 'TkAgg' by the following code:
  File "object_detection/model_main.py", line 26, in <module>
    from object_detection import model_lib
  File "/tensorflow/models/research/object_detection/model_lib.py", line 28, in <module>
    from object_detection import eval_util
  File "/tensorflow/models/research/object_detection/eval_util.py", line 28, in <module>
    from object_detection.metrics import coco_evaluation
  File "/tensorflow/models/research/object_detection/metrics/coco_evaluation.py", line 20, in <module>
    from object_detection.metrics import coco_tools
  File "/tensorflow/models/research/object_detection/metrics/coco_tools.py", line 47, in <module>
    from pycocotools import coco
  File "/tensorflow/models/research/pycocotools/coco.py", line 49, in <module>
    import matplotlib.pyplot as plt
  File "/usr/local/lib/python2.7/dist-packages/matplotlib/pyplot.py", line 71, in <module>
    from matplotlib.backends import pylab_setup
  File "/usr/local/lib/python2.7/dist-packages/matplotlib/backends/__init__.py", line 16, in <module>
    line for line in traceback.format_stack()


  import matplotlib; matplotlib.use('Agg')  # pylint: disable=multiple-statements
WARNING:tensorflow:Forced number of epochs for all eval validations to be 1.
WARNING:tensorflow:Expected number of evaluation epochs is 1, but instead encountered `eval_on_train_input_config.num_epochs` = 0. Overwriting `num_epochs` to 1.
WARNING:tensorflow:Estimator's model_fn (<function model_fn at 0x7f47116daaa0>) includes params argument, but params are not passed to Estimator.
WARNING:tensorflow:num_readers has been reduced to 10 to match input file shards.
WARNING:tensorflow:From /tensorflow/models/research/object_detection/builders/dataset_builder.py:86: parallel_interleave (from tensorflow.contrib.data.python.ops.interleave_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.data.experimental.parallel_interleave(...)`.
WARNING:tensorflow:From /usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/sparse_ops.py:1165: sparse_to_dense (from tensorflow.python.ops.sparse_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Create a `tf.sparse.SparseTensor` and use `tf.sparse.to_dense` instead.
WARNING:tensorflow:From /tensorflow/models/research/object_detection/builders/dataset_builder.py:158: batch_and_drop_remainder (from tensorflow.contrib.data.python.ops.batching) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.data.Dataset.batch(..., drop_remainder=True)`.
WARNING:root:Variable [BoxPredictor_0/ClassPredictor/biases] is available in checkpoint, but has an incompatible shape with model variable. Checkpoint shape: [[273]], model variable shape: [[9]]. This variable will not be initialized from the checkpoint.
WARNING:root:Variable [BoxPredictor_0/ClassPredictor/weights] is available in checkpoint, but has an incompatible shape with model variable. Checkpoint shape: [[1, 1, 512, 273]], model variable shape: [[1, 1, 512, 9]]. This variable will not be initialized from the checkpoint.
WARNING:root:Variable [BoxPredictor_1/ClassPredictor/biases] is available in checkpoint, but has an incompatible shape with model variable. Checkpoint shape: [[546]], model variable shape: [[18]]. This variable will not be initialized from the checkpoint.
WARNING:root:Variable [BoxPredictor_1/ClassPredictor/weights] is available in checkpoint, but has an incompatible shape with model variable. Checkpoint shape: [[1, 1, 1024, 546]], model variable shape: [[1, 1, 1024, 18]]. This variable will not be initialized from the checkpoint.
WARNING:root:Variable [BoxPredictor_2/ClassPredictor/biases] is available in checkpoint, but has an incompatible shape with model variable. Checkpoint shape: [[546]], model variable shape: [[18]]. This variable will not be initialized from the checkpoint.
WARNING:root:Variable [BoxPredictor_2/ClassPredictor/weights] is available in checkpoint, but has an incompatible shape with model variable. Checkpoint shape: [[1, 1, 512, 546]], model variable shape: [[1, 1, 512, 18]]. This variable will not be initialized from the checkpoint.
WARNING:root:Variable [BoxPredictor_3/ClassPredictor/biases] is available in checkpoint, but has an incompatible shape with model variable. Checkpoint shape: [[546]], model variable shape: [[18]]. This variable will not be initialized from the checkpoint.
WARNING:root:Variable [BoxPredictor_3/ClassPredictor/weights] is available in checkpoint, but has an incompatible shape with model variable. Checkpoint shape: [[1, 1, 256, 546]], model variable shape: [[1, 1, 256, 18]]. This variable will not be initialized from the checkpoint.
WARNING:root:Variable [BoxPredictor_4/ClassPredictor/biases] is available in checkpoint, but has an incompatible shape with model variable. Checkpoint shape: [[546]], model variable shape: [[18]]. This variable will not be initialized from the checkpoint.
WARNING:root:Variable [BoxPredictor_4/ClassPredictor/weights] is available in checkpoint, but has an incompatible shape with model variable. Checkpoint shape: [[1, 1, 256, 546]], model variable shape: [[1, 1, 256, 18]]. This variable will not be initialized from the checkpoint.
WARNING:root:Variable [BoxPredictor_5/ClassPredictor/biases] is available in checkpoint, but has an incompatible shape with model variable. Checkpoint shape: [[546]], model variable shape: [[18]]. This variable will not be initialized from the checkpoint.
WARNING:root:Variable [BoxPredictor_5/ClassPredictor/weights] is available in checkpoint, but has an incompatible shape with model variable. Checkpoint shape: [[1, 1, 128, 546]], model variable shape: [[1, 1, 128, 18]]. This variable will not be initialized from the checkpoint.
WARNING:root:Variable [global_step] is not available in checkpoint
2019-06-15 17:00:31.573478: W tensorflow/core/framework/allocator.cc:122] Allocation of 94633984 exceeds 10% of system memory.
2019-06-15 17:00:31.649518: W tensorflow/core/framework/allocator.cc:122] Allocation of 94633984 exceeds 10% of system memory.
2019-06-15 17:00:31.724917: W tensorflow/core/framework/allocator.cc:122] Allocation of 94633984 exceeds 10% of system memory.
2019-06-15 17:00:31.796480: W tensorflow/core/framework/allocator.cc:122] Allocation of 94633984 exceeds 10% of system memory.
2019-06-15 17:00:31.870867: W tensorflow/core/framework/allocator.cc:122] Allocation of 94633984 exceeds 10% of system memory.
./retrain_detection_model.sh: line 45:    44 Killed                  python object_detection/model_main.py --pipeline_config_path="${CKPT_DIR}/pipeline.config" --model_dir="${TRAIN_DIR}" --num_train_steps="${num_training_steps}" --num_eval_steps="${num_eval_steps}"

**Comment by avinash00721** [comment]
i have used :
NUM_TRAINING_STEPS=500 && \
NUM_EVAL_STEPS=100

and also 

./retrain_detection_model.sh \
--num_training_steps ${NUM_TRAINING_STEPS} \
--num_eval_steps ${NUM_EVAL_STEPS}

**Comment by jvishnuvardhan** [comment]
@avinash00721 If it is not related to bug/performance, then please post it in Stackoverflow. If you think it is a bug, then please provide more details on the issue and its context? Thanks!

**Comment by freemanls** [comment]
@jvishnuvardhan 

I am also very concerned about this issue because I have also encountered it.
Here is my background: I changed a model when I trained a model to 10,000 steps, which means retraining. When configuring all the items, running main.py is a problem.
What is the top-level directory of the model you are using: ssd_mobilenet_v1_coco_2018_01_28(300*300)
Did I write custom code (as opposed to using the inventory sample script provided in TensorFlow): basically no
OS platform and distribution (for example, Linux Ubuntu 16.04): Ubantu 19.04
TensorFlow installed from (source or binary): anaconda tensorflow-cpu 1.13.1
TensorFlow version (using the following command): 1.13.1
Bazel version (if compiled from source code): 0.24.1
CUDA / cuDNN version: 9/7
GPU model and memory: none
Accurate command reproduction:
Python model_main.py --model_dir=/home/leonliang/Progrom/01-Works/1906-TensorFlow-OB/training --pipeline_config_path=/home/leonliang/Progrom/01-Works/1906-TensorFlow-OB/training/pipeline .config --num_eval_steps=1 --num_train_steps=1 num_epochs=1 --alsologtostderr

------------------------------------ Next is the log
WARNING: The TensorFlow contrib module will not be included in TensorFlow 2.0.
For more information, please see:
  * https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md
  * https://github.com/tensorflow/addons
If you depend on functionality not listed there, please file an issue.

WARNING:tensorflow:Forced number of epochs for all eval validations to be 1.
WARNING:tensorflow: Expected number of evaluation epochs is 1, but instead encountered `eval_on_train_input_config.num_epochs` = 0. Overwriting `num_epochs` to 1.
WARNING:tensorflow:Estimator's model_fn (<function create_model_fn.<locals>.model_fn at 0x7f8e5752e048>) includes params argument, but params are not passed to Estimator.

--------------------------------------- End of log
The program stopped here, and I didn't do it when I was training.

I hope to get a reply.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be an old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
